Toolchain US 35 responder

AppaYY opened the pull request:
Moet nog goedgekeurd worden door de beta testers

Commit:
Added a new responder called All tags responder

Commit:
Fixed AllTagsResponder

Commit:
Started with Unit Test

Commit:
Unit test update

Commit:
Added a check for duplicates and deleted unnecessary files

Commit:
Changed variable names in test

Commit:
Changed name of test variables

Commit:
Removed unnecessary imports in test & renamed functions and variables

Commit:
Fixed issue

Commit:
Adjusted testcase name and List type & added own WikiPageDummy class in testcase.

Commit:
Separated Dummy file and made first assertion

Commit:
Set property attempt 2

Commit:
Fixed issue, adjusted Dummy class name and added documentation

Commit:
Removed unnecessary variables & optimized test code

Commit:
Added children tag test

Commit:
Fixed warnings

Commit:
Added pom.xml to gitignore

AppaYY committed:
Update pom.xml

Commit:
Adjusted test case names & import classes

Commit:
Added new test

AppaYY committed:
Update pom.xml

Commit:
Removed pom.xml from gitignore & updated pom.xml

Commit:
Merge remote-tracking branch 'origin/Toolchain-US-35-Responder' into Toolchain-US-35-Responder

tcnh commented on src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    private List<String> getPageTagsHelper(SourcePage page, List<String> allTagsArray) {
+        String[] tags = page.getProperty(WikiPageProperty.SUITES).split(", ");
```
suggestie: `.split("\\s*,\\s*")` om niet tegen problemen aan te lopen als er meer of minder whitespace om de komma's zit

AppaYY replied:
Split is aangepast.

maartenjanvangool replied:
@tcnh ik heb tegen ze gezegd dat ze het zo moeten doen omdat je dat 2e argument alleen nodig hebt voor de recursie, iets waar de buitenwereld niks mee te maken heeft. daarom dat die helperfunctie ook private is.

tcnh commented on src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java:
```diff
@@ -0,0 +1,80 @@
+        return response;
+    }
+
+    public List<String> getPageTags(SourcePage page){
```
Zou dit niet een Set moeten zijn (je wil geen dubbele entries toch?)

AppaYY replied:
Ik wil inderdaad geen dubbele entries, Set geïmplementeerd.

tcnh commented on src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java:
```diff
@@ -0,0 +1,80 @@
+        return page;
+    }
+
+    private List<String> getPageTagsHelper(SourcePage page, List<String> allTagsArray) {
```
Deze functie geeft geen 'helper' terug, maar een lijstje. Ik zou 'm hernoemen naar iets van `allTagsForPage` oid?

AppaYY replied:
Naam aangepast naar `getPageTags`

tcnh commented on src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java:
```diff
@@ -0,0 +1,80 @@
+        }
+
+        return allTagsArray;
+    }
```
Suggestie voor kortere functie:
```
private Set<String> allTagsForPage(SourcePage page, Set<String> allTags) {
        String[] tags = page.getProperty(WikiPageProperty.SUITES).split("\\s*,\\s*");
        allTags.addAll(Arrays.asList(tags));

        for (SourcePage p : getSortedChildren(page)) {
            allTagsForPage(p, allTags);
        }

        return allTags;
    }
```

AppaYY replied:
Ik heb de functie geprobeerd te implementeren, maar wanneer de childpage geen tag had kreeg ik gegarandeerd een lege value terug `"", "mainPageTag"`. 

Ik heb mijn originele functie hierdoor zo gehouden.

tcnh commented on src/main/java/nl/praegus/fitnesse/responders/allTags/TestWikiPageDummy.java:
```diff
@@ -0,0 +1,65 @@
+import java.util.Collections;
+import java.util.List;
+
+class TestWikiPageDummy extends BaseWikiPage {
```
Waarom extend je niet fitnesse.wiki.WikipageDummy ipv hem 90% te kopieren? Dan hoef je alleen je constructors te maken en getChildren te overriden. De rest van de impl zit al in WikipageDummy ?

tcnh replied:
En zou deze klasse niet in test moeten leven ipv in main?

AppaYY replied:
Ik heb hem nu ge-extend en de benodigde functies override. Klasse zit nu ook in de test directory in plaats van de main directory.

tcnh commented on src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    public List<String> getPageTags(SourcePage page){
+        return getPageTagsHelper(page, new ArrayList<>());
```
Waarom deze tussenstap? De logica uit de getPageTagsHelper kan toch gewoon hier leven?

AppaYY replied:
De tussenstap was inderdaad onnodig. Ik heb ze nu samengevoegd.

AppaYY replied:
Na het gesprek met Maarten-Jan, blijkt het toch wel beter te zijn om de tussenstap te hebben met getPageTagshelper.

Lizzze committed:
Merge pull request #11 from praegus/master

mergen met master

PandoraMedusa committed:
Review changes added (Comments 1 & 3)

PandoraMedusa committed:
Fixed placement of wikiPageDummy file

PandoraMedusa committed:
Comment 5

Commit:
Removed unnecessary comments, fixed review comments & adjusted test cases

Commit:
Fixed test case issue

Commit:
Reverted function logic & adjusted test cases

PandoraMedusa committed:
Merge branch 'master' into Toolchain-US-35-Responder

maartenjanvangool commented on src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java:
```diff
@@ -0,0 +1,77 @@
+        return result;
+    }
+
+    private WikiPage loadPage(FitNesseContext context, String pageName, Map<String, String> inputs) {
```
Dit stuk code is dubbel met de Table of contenst responder. Misschien een klasse toevoegen en deze functie static maken?

AppaYY replied:
Globale klasse toegevoegd voor loadPage.

maartenjanvangool commented on src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java:
```diff
@@ -0,0 +1,77 @@
+
+    public Set<String> getPageTags(SourcePage page, Set<String> allTags){
+        String[] tags = page.getProperty(WikiPageProperty.SUITES).split("\\s*,\\s*");
+        for(String tag: tags) {
```
deze hele loop kan vervangen worden door een mooie stream volgens mij: (snap dat dit wat geavanceerd is :-)) 
allTags.addAll(Arrays.stream(tags).filter(t -> t.length()>0).collect(Collectors.toList()));

AppaYY replied:
Stream geimplementeerd. 

Moest dit wel even goed bestuderen, maar het is zeker handiger dan een for loop.

maartenjanvangool commented on src/main/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponder.java:
```diff
@@ -0,0 +1,77 @@
+        return new SecureReadOperation();
+    }
+
+    private List<SourcePage> getSortedChildren(SourcePage parent) {
```
waarom sorteren we die dingen?

AppaYY replied:
De sorting is weggehaald. Sorteren is onnodig voor deze responder, aangezien het niet uitmaakt in welke volgorde de tags zijn.

maartenjanvangool commented on src/test/java/nl/praegus/fitnesse/responders/allTags/AllTagsResponderTest.java:
```diff
@@ -0,0 +1,75 @@
+        assertThat(receivedValue).isEqualTo(new HashSet<>(Collections.singletonList("mainTag")));
+    }
+
+    /* getPageTags */
```
dit commentaar is nutteloos, moet overal weg

AppaYY replied:
Commentaar weggehaald.

Commit:
Set back GetPageTagsHelper & removed unnecessary comments

Commit:
Optimized getPageTagsHelper tag loop & removed sorting of pagechildren

Commit:
Adjusted Table of contents responder variable names

Commit:
Added a WikiPageHelper class to optimize methods to prevent duplicates

AppaYY commented:
Variabele van Table of contents responder aangepast voor meer leesbaarheid.

tcnh committed:
table decorator

tcnh committed:
Table decorator

tcnh committed:
Debug script now only needs hsac fixtures on classpath
Timeout defaults to 10 mins (600K ms), can be set using fitnesse.debugscript.timeout sysprop

tcnh committed:
revert pom update
